.github/workflows/deploy-dashboard.yaml

```diff
@@ -25,7 +25,7 @@ jobs:
       id-token: write
     steps:
       - name: Checkout
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
 
       - name: Setup env
         uses: ./.github/actions/common/setup-env
```

Comment:
have some other scripts to double check here https://github.com/patrick-rivos/gcc-postcommit-ci/blob/main/.github/workflows/deploy-dashboard.yaml#L73-L90

Reply:
EndBug/add-and-commit@v9 does not have a v10
actions/deploy-pages@v2 Updated to v4
actions/configure-pages@v3 seems fine?
actions/upload-pages-artifact@v2 seems fine?

Reply:
actions/configure-pages looks like theres a v5 https://github.com/actions/configure-pages/releases. the README hasn't been updated in 2 years but the action was recently updated to use node 20 (https://github.com/actions/configure-pages/commit/3a014131e99a7bf2daf896787a8f92789126d038)

actions/upload-pages-artifact looks like theres a v4 https://github.com/actions/upload-pages-artifact/releases/tag/v3.0.0. Not sure what the difference is bc the changelog doesn't explicitly say it changes to use node 20